diff --unified --recursive --new-file before/github/NamedUser.py after/github/NamedUser.py
--- before/github/NamedUser.py	YYYY-MM-DD HH:MM:SS +0000
+++ after/github/NamedUser.py	YYYY-MM-DD HH:MM:SS +0000
@@ -40,6 +40,7 @@
 
     The OpenAPI schema can be found at
 
+    - /components/schemas/nullable-simple-user
     - /components/schemas/public-user
 
     """
diff --unified --recursive --new-file before/openapi.index after/openapi.index
--- before/openapi.index	YYYY-MM-DD HH:MM:SS +0000
+++ after/openapi.index	YYYY-MM-DD HH:MM:SS +0000
@@ -443,7 +443,7 @@
       "bases": [
         "CompletableGithubObject"
       ],
-      "docstring": "This class represents NamedUser.\n\n    The reference can be found here\n    https://docs.github.com/en/rest/reference/users#get-a-user\n\n    The OpenAPI schema can be found at\n\n    - /components/schemas/public-user",
+      "docstring": "This class represents NamedUser.\n\n    The reference can be found here\n    https://docs.github.com/en/rest/reference/users#get-a-user\n\n    The OpenAPI schema can be found at\n\n    - /components/schemas/nullable-simple-user\n    - /components/schemas/public-user",
       "filename": "github/NamedUser.py",
       "ids": [
         "some_attribute"
@@ -464,11 +464,13 @@
           "returns": [
             "str"
           ],
-          "schema": {
-            "format": "uri",
-            "type": "string"
-          },
+          "schema": null,
           "schemas": {
+            "/components/schemas/nullable-simple-user": {
+              "example": "https://github.com/images/error/octocat_happy.gif",
+              "format": "uri",
+              "type": "string"
+            },
             "/components/schemas/public-user": {
               "format": "uri",
               "type": "string"
@@ -580,12 +582,12 @@
             "str",
             "None"
           ],
-          "schema": {
-            "format": "email",
-            "nullable": true,
-            "type": "string"
-          },
+          "schema": null,
           "schemas": {
+            "/components/schemas/nullable-simple-user": {
+              "nullable": true,
+              "type": "string"
+            },
             "/components/schemas/public-user": {
               "format": "email",
               "nullable": true,
@@ -598,10 +600,12 @@
           "returns": [
             "str"
           ],
-          "schema": {
-            "type": "string"
-          },
+          "schema": null,
           "schemas": {
+            "/components/schemas/nullable-simple-user": {
+              "example": "https://api.github.com/users/octocat/events{/privacy}",
+              "type": "string"
+            },
             "/components/schemas/public-user": {
               "type": "string"
             }
@@ -626,11 +630,13 @@
           "returns": [
             "str"
           ],
-          "schema": {
-            "format": "uri",
-            "type": "string"
-          },
+          "schema": null,
           "schemas": {
+            "/components/schemas/nullable-simple-user": {
+              "example": "https://api.github.com/users/octocat/followers",
+              "format": "uri",
+              "type": "string"
+            },
             "/components/schemas/public-user": {
               "format": "uri",
               "type": "string"
@@ -656,10 +662,12 @@
           "returns": [
             "str"
           ],
-          "schema": {
-            "type": "string"
-          },
+          "schema": null,
           "schemas": {
+            "/components/schemas/nullable-simple-user": {
+              "example": "https://api.github.com/users/octocat/following{/other_user}",
+              "type": "string"
+            },
             "/components/schemas/public-user": {
               "type": "string"
             }
@@ -670,10 +678,12 @@
           "returns": [
             "str"
           ],
-          "schema": {
-            "type": "string"
-          },
+          "schema": null,
           "schemas": {
+            "/components/schemas/nullable-simple-user": {
+              "example": "https://api.github.com/users/octocat/gists{/gist_id}",
+              "type": "string"
+            },
             "/components/schemas/public-user": {
               "type": "string"
             }
@@ -685,11 +695,13 @@
             "str",
             "None"
           ],
-          "schema": {
-            "nullable": true,
-            "type": "string"
-          },
+          "schema": null,
           "schemas": {
+            "/components/schemas/nullable-simple-user": {
+              "example": "41d064eb2195891e12d0413f63227ea7",
+              "nullable": true,
+              "type": "string"
+            },
             "/components/schemas/public-user": {
               "nullable": true,
               "type": "string"
@@ -718,11 +730,13 @@
           "returns": [
             "str"
           ],
-          "schema": {
-            "format": "uri",
-            "type": "string"
-          },
+          "schema": null,
           "schemas": {
+            "/components/schemas/nullable-simple-user": {
+              "example": "https://github.com/octocat",
+              "format": "uri",
+              "type": "string"
+            },
             "/components/schemas/public-user": {
               "format": "uri",
               "type": "string"
@@ -734,11 +748,13 @@
           "returns": [
             "int"
           ],
-          "schema": {
-            "format": "int64",
-            "type": "integer"
-          },
+          "schema": null,
           "schemas": {
+            "/components/schemas/nullable-simple-user": {
+              "example": 1,
+              "format": "int64",
+              "type": "integer"
+            },
             "/components/schemas/public-user": {
               "format": "int64",
               "type": "integer"
@@ -767,10 +783,12 @@
           "returns": [
             "str"
           ],
-          "schema": {
-            "type": "string"
-          },
+          "schema": null,
           "schemas": {
+            "/components/schemas/nullable-simple-user": {
+              "example": "octocat",
+              "type": "string"
+            },
             "/components/schemas/public-user": {
               "type": "string"
             }
@@ -787,6 +805,10 @@
             "type": "string"
           },
           "schemas": {
+            "/components/schemas/nullable-simple-user": {
+              "nullable": true,
+              "type": "string"
+            },
             "/components/schemas/public-user": {
               "nullable": true,
               "type": "string"
@@ -798,10 +820,12 @@
           "returns": [
             "str"
           ],
-          "schema": {
-            "type": "string"
-          },
+          "schema": null,
           "schemas": {
+            "/components/schemas/nullable-simple-user": {
+              "example": "MDQ6VXNlcjE=",
+              "type": "string"
+            },
             "/components/schemas/public-user": {
               "type": "string"
             }
@@ -831,11 +855,13 @@
           "returns": [
             "str"
           ],
-          "schema": {
-            "format": "uri",
-            "type": "string"
-          },
+          "schema": null,
           "schemas": {
+            "/components/schemas/nullable-simple-user": {
+              "example": "https://api.github.com/users/octocat/orgs",
+              "format": "uri",
+              "type": "string"
+            },
             "/components/schemas/public-user": {
               "format": "uri",
               "type": "string"
@@ -907,11 +933,13 @@
           "returns": [
             "str"
           ],
-          "schema": {
-            "format": "uri",
-            "type": "string"
-          },
+          "schema": null,
           "schemas": {
+            "/components/schemas/nullable-simple-user": {
+              "example": "https://api.github.com/users/octocat/received_events",
+              "format": "uri",
+              "type": "string"
+            },
             "/components/schemas/public-user": {
               "format": "uri",
               "type": "string"
@@ -923,11 +951,13 @@
           "returns": [
             "str"
           ],
-          "schema": {
-            "format": "uri",
-            "type": "string"
-          },
+          "schema": null,
           "schemas": {
+            "/components/schemas/nullable-simple-user": {
+              "example": "https://api.github.com/users/octocat/repos",
+              "format": "uri",
+              "type": "string"
+            },
             "/components/schemas/public-user": {
               "format": "uri",
               "type": "string"
@@ -943,6 +973,9 @@
             "type": "boolean"
           },
           "schemas": {
+            "/components/schemas/nullable-simple-user": {
+              "type": "boolean"
+            },
             "/components/schemas/public-user": {
               "type": "boolean"
             }
@@ -953,10 +986,12 @@
           "returns": [
             "str"
           ],
-          "schema": {
-            "type": "string"
-          },
+          "schema": null,
           "schemas": {
+            "/components/schemas/nullable-simple-user": {
+              "example": "https://api.github.com/users/octocat/starred{/owner}{/repo}",
+              "type": "string"
+            },
             "/components/schemas/public-user": {
               "type": "string"
             }
@@ -967,11 +1002,13 @@
           "returns": [
             "str"
           ],
-          "schema": {
-            "format": "uri",
-            "type": "string"
-          },
+          "schema": null,
           "schemas": {
+            "/components/schemas/nullable-simple-user": {
+              "example": "https://api.github.com/users/octocat/subscriptions",
+              "format": "uri",
+              "type": "string"
+            },
             "/components/schemas/public-user": {
               "format": "uri",
               "type": "string"
@@ -1016,10 +1053,12 @@
           "returns": [
             "str"
           ],
-          "schema": {
-            "type": "string"
-          },
+          "schema": null,
           "schemas": {
+            "/components/schemas/nullable-simple-user": {
+              "example": "User",
+              "type": "string"
+            },
             "/components/schemas/public-user": {
               "type": "string"
             }
@@ -1046,10 +1085,12 @@
           "returns": [
             "str"
           ],
-          "schema": {
-            "type": "string"
-          },
+          "schema": null,
           "schemas": {
+            "/components/schemas/nullable-simple-user": {
+              "example": "public",
+              "type": "string"
+            },
             "/components/schemas/public-user": {
               "type": "string"
             }
@@ -1057,6 +1098,7 @@
         }
       },
       "schemas": [
+        "/components/schemas/nullable-simple-user",
         "/components/schemas/public-user"
       ],
       "test_filename": "tests/NamedUser.py"
@@ -1077,6 +1119,7 @@
         "NamedUser"
       ],
       "inherited_schemas": [
+        "/components/schemas/nullable-simple-user",
         "/components/schemas/public-user"
       ],
       "methods": {},
@@ -1898,6 +1941,9 @@
       "/components/schemas/issue-comment": [
         "IssueComment"
       ],
+      "/components/schemas/nullable-simple-user": [
+        "NamedUser"
+      ],
       "/components/schemas/public-user": [
         "NamedUser"
       ],
