diff --unified --recursive --new-file before/github/NamedUser.py after/github/NamedUser.py
--- before/github/NamedUser.py	YYYY-MM-DD HH:MM:SS +0000
+++ after/github/NamedUser.py	YYYY-MM-DD HH:MM:SS +0000
@@ -42,6 +42,7 @@
 
     - /components/schemas/nullable-simple-user
     - /components/schemas/public-user
+    - /components/schemas/simple-user
 
     """
 
diff --unified --recursive --new-file before/openapi.index after/openapi.index
--- before/openapi.index	YYYY-MM-DD HH:MM:SS +0000
+++ after/openapi.index	YYYY-MM-DD HH:MM:SS +0000
@@ -466,7 +466,7 @@
       "bases": [
         "CompletableGithubObject"
       ],
-      "docstring": "This class represents NamedUser.\n\n    The reference can be found here\n    https://docs.github.com/en/rest/reference/users#get-a-user\n\n    The OpenAPI schema can be found at\n\n    - /components/schemas/nullable-simple-user\n    - /components/schemas/public-user",
+      "docstring": "This class represents NamedUser.\n\n    The reference can be found here\n    https://docs.github.com/en/rest/reference/users#get-a-user\n\n    The OpenAPI schema can be found at\n\n    - /components/schemas/nullable-simple-user\n    - /components/schemas/public-user\n    - /components/schemas/simple-user",
       "filename": "github/NamedUser.py",
       "ids": [
         "some_attribute"
@@ -509,6 +509,11 @@
             "/components/schemas/public-user": {
               "format": "uri",
               "type": "string"
+            },
+            "/components/schemas/simple-user": {
+              "example": "https://github.com/images/error/octocat_happy.gif",
+              "format": "uri",
+              "type": "string"
             }
           }
         },
@@ -627,6 +632,10 @@
               "format": "email",
               "nullable": true,
               "type": "string"
+            },
+            "/components/schemas/simple-user": {
+              "nullable": true,
+              "type": "string"
             }
           }
         },
@@ -643,6 +652,10 @@
             },
             "/components/schemas/public-user": {
               "type": "string"
+            },
+            "/components/schemas/simple-user": {
+              "example": "https://api.github.com/users/octocat/events{/privacy}",
+              "type": "string"
             }
           }
         },
@@ -675,6 +688,11 @@
             "/components/schemas/public-user": {
               "format": "uri",
               "type": "string"
+            },
+            "/components/schemas/simple-user": {
+              "example": "https://api.github.com/users/octocat/followers",
+              "format": "uri",
+              "type": "string"
             }
           }
         },
@@ -705,6 +723,10 @@
             },
             "/components/schemas/public-user": {
               "type": "string"
+            },
+            "/components/schemas/simple-user": {
+              "example": "https://api.github.com/users/octocat/following{/other_user}",
+              "type": "string"
             }
           }
         },
@@ -721,6 +743,10 @@
             },
             "/components/schemas/public-user": {
               "type": "string"
+            },
+            "/components/schemas/simple-user": {
+              "example": "https://api.github.com/users/octocat/gists{/gist_id}",
+              "type": "string"
             }
           }
         },
@@ -740,6 +766,11 @@
             "/components/schemas/public-user": {
               "nullable": true,
               "type": "string"
+            },
+            "/components/schemas/simple-user": {
+              "example": "41d064eb2195891e12d0413f63227ea7",
+              "nullable": true,
+              "type": "string"
             }
           }
         },
@@ -775,6 +806,11 @@
             "/components/schemas/public-user": {
               "format": "uri",
               "type": "string"
+            },
+            "/components/schemas/simple-user": {
+              "example": "https://github.com/octocat",
+              "format": "uri",
+              "type": "string"
             }
           }
         },
@@ -793,6 +829,11 @@
             "/components/schemas/public-user": {
               "format": "int64",
               "type": "integer"
+            },
+            "/components/schemas/simple-user": {
+              "example": 1,
+              "format": "int64",
+              "type": "integer"
             }
           }
         },
@@ -826,6 +867,10 @@
             },
             "/components/schemas/public-user": {
               "type": "string"
+            },
+            "/components/schemas/simple-user": {
+              "example": "octocat",
+              "type": "string"
             }
           }
         },
@@ -847,6 +892,10 @@
             "/components/schemas/public-user": {
               "nullable": true,
               "type": "string"
+            },
+            "/components/schemas/simple-user": {
+              "nullable": true,
+              "type": "string"
             }
           }
         },
@@ -863,6 +912,10 @@
             },
             "/components/schemas/public-user": {
               "type": "string"
+            },
+            "/components/schemas/simple-user": {
+              "example": "MDQ6VXNlcjE=",
+              "type": "string"
             }
           }
         },
@@ -900,6 +953,11 @@
             "/components/schemas/public-user": {
               "format": "uri",
               "type": "string"
+            },
+            "/components/schemas/simple-user": {
+              "example": "https://api.github.com/users/octocat/orgs",
+              "format": "uri",
+              "type": "string"
             }
           }
         },
@@ -978,6 +1036,11 @@
             "/components/schemas/public-user": {
               "format": "uri",
               "type": "string"
+            },
+            "/components/schemas/simple-user": {
+              "example": "https://api.github.com/users/octocat/received_events",
+              "format": "uri",
+              "type": "string"
             }
           }
         },
@@ -996,6 +1059,11 @@
             "/components/schemas/public-user": {
               "format": "uri",
               "type": "string"
+            },
+            "/components/schemas/simple-user": {
+              "example": "https://api.github.com/users/octocat/repos",
+              "format": "uri",
+              "type": "string"
             }
           }
         },
@@ -1013,6 +1081,9 @@
             },
             "/components/schemas/public-user": {
               "type": "boolean"
+            },
+            "/components/schemas/simple-user": {
+              "type": "boolean"
             }
           }
         },
@@ -1029,6 +1100,10 @@
             "/components/schemas/nullable-simple-user": {
               "example": "\"2020-07-09T00:17:55Z\"",
               "type": "string"
+            },
+            "/components/schemas/simple-user": {
+              "example": "\"2020-07-09T00:17:55Z\"",
+              "type": "string"
             }
           }
         },
@@ -1045,6 +1120,10 @@
             },
             "/components/schemas/public-user": {
               "type": "string"
+            },
+            "/components/schemas/simple-user": {
+              "example": "https://api.github.com/users/octocat/starred{/owner}{/repo}",
+              "type": "string"
             }
           }
         },
@@ -1063,6 +1142,11 @@
             "/components/schemas/public-user": {
               "format": "uri",
               "type": "string"
+            },
+            "/components/schemas/simple-user": {
+              "example": "https://api.github.com/users/octocat/subscriptions",
+              "format": "uri",
+              "type": "string"
             }
           }
         },
@@ -1112,6 +1196,10 @@
             },
             "/components/schemas/public-user": {
               "type": "string"
+            },
+            "/components/schemas/simple-user": {
+              "example": "User",
+              "type": "string"
             }
           }
         },
@@ -1144,13 +1232,18 @@
             },
             "/components/schemas/public-user": {
               "type": "string"
+            },
+            "/components/schemas/simple-user": {
+              "example": "public",
+              "type": "string"
             }
           }
         }
       },
       "schemas": [
         "/components/schemas/nullable-simple-user",
-        "/components/schemas/public-user"
+        "/components/schemas/public-user",
+        "/components/schemas/simple-user"
       ],
       "test_filename": "tests/NamedUser.py"
     },
@@ -1171,7 +1264,8 @@
       ],
       "inherited_schemas": [
         "/components/schemas/nullable-simple-user",
-        "/components/schemas/public-user"
+        "/components/schemas/public-user",
+        "/components/schemas/simple-user"
       ],
       "methods": {},
       "module": "NamedUser",
@@ -3808,6 +3902,9 @@
       "/components/schemas/repository": [
         "Repository"
       ],
+      "/components/schemas/simple-user": [
+        "NamedUser"
+      ],
       "/components/schemas/user-search-result-item": [
         "NamedUserSearchResult"
       ]
