diff --unified --recursive --new-file before/github/Commit.py after/github/Commit.py
--- before/github/Commit.py	YYYY-MM-DD HH:MM:SS +0000
+++ after/github/Commit.py	YYYY-MM-DD HH:MM:SS +0000
@@ -60,11 +60,15 @@
         self._comments_url: Attribute[str] = NotSet
         self._commit: Attribute[GitCommit] = NotSet
         self._committer: Attribute[NamedUser] = NotSet
+        self._files: Attribute[list[dict[str, Any]]] = NotSet
         self._html_url: Attribute[str] = NotSet
         self._message: Attribute[str] = NotSet
         self._node_id: Attribute[str] = NotSet
         self._parents: Attribute[list[Commit]] = NotSet
         self._sha: Attribute[str] = NotSet
+        self._stats: Attribute[dict[str, Any]] = NotSet
+        self._tree: Attribute[dict[str, Any]] = NotSet
+        self._verification: Attribute[dict[str, Any]] = NotSet
 
     def __repr__(self) -> str:
         # TODO: replace "some_attribute" with uniquely identifying attributes in the dict, then run:
@@ -96,6 +100,11 @@
         return self._committer.value
 
     @property
+    def files(self) -> list[dict[str, Any]]:
+        self._completeIfNotSet(self._files)
+        return self._files.value
+
+    @property
     def html_url(self) -> str:
         self._completeIfNotSet(self._html_url)
         return self._html_url.value
@@ -120,6 +129,21 @@
         self._completeIfNotSet(self._sha)
         return self._sha.value
 
+    @property
+    def stats(self) -> dict[str, Any]:
+        self._completeIfNotSet(self._stats)
+        return self._stats.value
+
+    @property
+    def tree(self) -> dict[str, Any]:
+        self._completeIfNotSet(self._tree)
+        return self._tree.value
+
+    @property
+    def verification(self) -> dict[str, Any]:
+        self._completeIfNotSet(self._verification)
+        return self._verification.value
+
     def _useAttributes(self, attributes: dict[str, Any]) -> None:
         super()._useAttributes(attributes)
         if "author" in attributes:  # pragma no branch
@@ -132,6 +156,8 @@
             self._commit = self._makeClassAttribute(github.GitCommit.GitCommit, attributes["commit"])
         if "committer" in attributes:  # pragma no branch
             self._committer = self._makeClassAttribute(github.NamedUser.NamedUser, attributes["committer"])
+        if "files" in attributes:  # pragma no branch
+            self._files = self._makeListOfDictsAttribute(attributes["files"])
         if "html_url" in attributes:  # pragma no branch
             self._html_url = self._makeStringAttribute(attributes["html_url"])
         if "message" in attributes:  # pragma no branch
@@ -142,4 +168,10 @@
             self._parents = self._makeListOfClassesAttribute(github.Commit.Commit, attributes["parents"])
         if "sha" in attributes:  # pragma no branch
             self._sha = self._makeStringAttribute(attributes["sha"])
+        if "stats" in attributes:  # pragma no branch
+            self._stats = self._makeDictAttribute(attributes["stats"])
+        if "tree" in attributes:  # pragma no branch
+            self._tree = self._makeDictAttribute(attributes["tree"])
+        if "verification" in attributes:  # pragma no branch
+            self._verification = self._makeDictAttribute(attributes["verification"])
 
diff --unified --recursive --new-file before/openapi.index after/openapi.index
--- before/openapi.index	YYYY-MM-DD HH:MM:SS +0000
+++ after/openapi.index	YYYY-MM-DD HH:MM:SS +0000
@@ -258,6 +258,26 @@
             }
           }
         },
+        "files": {
+          "name": "files",
+          "returns": [
+            "list[dict[str, Any]]"
+          ],
+          "schema": {
+            "items": {
+              "$ref": "#/components/schemas/diff-entry"
+            },
+            "type": "array"
+          },
+          "schemas": {
+            "/components/schemas/commit": {
+              "items": {
+                "$ref": "#/components/schemas/diff-entry"
+              },
+              "type": "array"
+            }
+          }
+        },
         "html_url": {
           "name": "html_url",
           "returns": [
@@ -389,6 +409,96 @@
               "type": "string"
             }
           }
+        },
+        "stats": {
+          "name": "stats",
+          "returns": [
+            "dict[str, Any]"
+          ],
+          "schema": {
+            "properties": {
+              "additions": {
+                "type": "integer"
+              },
+              "deletions": {
+                "type": "integer"
+              },
+              "total": {
+                "type": "integer"
+              }
+            },
+            "type": "object"
+          },
+          "schemas": {
+            "/components/schemas/commit": {
+              "properties": {
+                "additions": {
+                  "type": "integer"
+                },
+                "deletions": {
+                  "type": "integer"
+                },
+                "total": {
+                  "type": "integer"
+                }
+              },
+              "type": "object"
+            }
+          }
+        },
+        "tree": {
+          "name": "tree",
+          "returns": [
+            "dict[str, Any]"
+          ],
+          "schema": {
+            "properties": {
+              "sha": {
+                "type": "string"
+              },
+              "url": {
+                "format": "uri",
+                "type": "string"
+              }
+            },
+            "required": [
+              "sha",
+              "url"
+            ],
+            "type": "object"
+          },
+          "schemas": {
+            "/components/schemas/commit-search-result-item/properties/commit": {
+              "properties": {
+                "sha": {
+                  "type": "string"
+                },
+                "url": {
+                  "format": "uri",
+                  "type": "string"
+                }
+              },
+              "required": [
+                "sha",
+                "url"
+              ],
+              "type": "object"
+            }
+          }
+        },
+        "verification": {
+          "name": "verification",
+          "returns": [
+            "dict[str, Any]"
+          ],
+          "schema": {
+            "$ref": "#/components/schemas/verification"
+          },
+          "schemas": {
+            "/components/schemas/commit-search-result-item/properties/commit": {
+              "$ref": "#/components/schemas/verification"
+            }
+          }
         }
       },
       "schemas": [
diff --unified --recursive --new-file before/tests/Commit.py after/tests/Commit.py
--- before/tests/Commit.py	YYYY-MM-DD HH:MM:SS +0000
+++ after/tests/Commit.py	YYYY-MM-DD HH:MM:SS +0000
@@ -43,9 +43,13 @@
         self.assertEqual(c.comments_url, "")
         self.assertEqual(c.commit.some_attribute, "")
         self.assertEqual(c.committer.some_attribute, "")
+        self.assertEqual(c.files, "list[dict[str, Any]]")
         self.assertEqual(c.html_url, "")
         self.assertEqual(c.message, "")
         self.assertEqual(c.node_id, "")
         self.assertEqual(c.parents[0].some_attribute, "")
         self.assertEqual(c.sha, "")
+        self.assertEqual(c.stats, "dict[str, Any]")
+        self.assertEqual(c.tree, "dict[str, Any]")
         self.assertEqual(c.url, "")
+        self.assertEqual(c.verification, "dict[str, Any]")
