--- github/NamedUser.py.orig	2026-06-03 11:00:10.107879932 +0200
+++ github/NamedUser.py.patched	2026-06-03 11:04:46.027307968 +0200
@@ -24,11 +24,13 @@
 from typing import Any, TYPE_CHECKING
 from datetime import datetime, timezone
 
+import github.Repository
+from github.PaginatedList import PaginatedList
 from github.GithubObject import CompletableGithubObject
-from github.GithubObject import Attribute, NotSet
+from github.GithubObject import Attribute, NotSet, is_optional, method_parameter, openapi_parameter, Opt
 
 if TYPE_CHECKING:
-    pass
+    from github.Repository import Repository
 
 
 class NamedUser(CompletableGithubObject):
@@ -286,7 +288,49 @@
     def user_view_type(self) -> str:
         self._completeIfNotSet(self._user_view_type)
         return self._user_view_type.value
-    
+
+    @method_parameter(
+        name="name",
+        required=True,
+        merge=["owner", "repo"],
+        docstring_prepend='The name of the repository in the form "{owner}/{repo}" with:\n',
+    )
+    @openapi_parameter(name="owner", input=True, docstring_prepend='- "{owner}": ')
+    @openapi_parameter(name="repo", input=True, docstring_prepend='- "{repo}": ')
+    def get_repo(self, name: str) -> Repository:
+        """
+        :calls: `GET /repos/{owner}/{repo} <https://docs.github.com/en/rest/reference/repos>`_
+        """
+        assert isinstance(name, str), name
+        url = f"/repos/{self.login}/{name}"
+        return github.Repository.Repository(self._requester, url=url)
+
+    def get_repos(
+        self,
+        type: Opt[str] = NotSet,
+        sort: Opt[str] = NotSet,
+        direction: Opt[str] = NotSet,
+    ) -> PaginatedList[Repository]:
+        """
+        :calls: `GET /users/{username}/repos <https://docs.github.com/en/rest/reference/repos>`_
+        """
+        assert type is NotSet or isinstance(type, str), type
+        assert sort is NotSet or isinstance(sort, str), sort
+        assert direction is NotSet or isinstance(direction, str), direction
+        url_parameters = dict()
+        if type is not NotSet:
+            url_parameters["type"] = type
+        if sort is not NotSet:
+            url_parameters["sort"] = sort
+        if direction is not NotSet:
+            url_parameters["direction"] = direction
+        return github.PaginatedList.PaginatedList(
+            github.Repository.Repository,
+            self._requester,
+            f"{self.url}/repos",
+            url_parameters,
+        )
+
     def get_starred(self, sort: Opt[str] = NotSet, direction: Opt[str] = NotSet) -> None:
         """
         List repositories starred by a user
